ArduCopter/mode_auto.cpp

```diff
@@ -1912,17 +1997,7 @@ bool ModeAuto::verify_nav_wp(const AP_Mission::Mission_Command& cmd)
 // verify_circle - check if we have circled the point enough
 bool ModeAuto::verify_circle(const AP_Mission::Mission_Command& cmd)
 {
-    // check if we've reached the edge
-    if (mode() == SubMode::CIRCLE_MOVE_TO_EDGE) {
-        if (copter.wp_nav->reached_wp_destination()) {
-            // start circling
-            circle_start();
-        }
-        return false;
-    }
-
-    // check if we have completed circling
-    return fabsf(copter.circle_nav->get_angle_total()/M_2PI) >= LOWBYTE(cmd.p1);
+    return loiter_turns.is_completed();
```

Comment:
The `is_completed()` call could probably stay - but the update() call should be made in modeauto::run, surely?